### PR TITLE
adding tilegrid_inflator

### PR DESCRIPTION
**Pull request by FoamyGuy**
New helper function for inflating tilegrids, essentially helper code for what is shown on this learn guide page: https://learn.adafruit.com/circuitpython-display-support-using-displayio/multiple-tilegrids

Marking this as Draft for now, because I think it should still get an example added before merging. But I'm open to review / feedback if anyone wants to take a look.

### CODE REVIEW

**Comment by FoamyGuy** [comment]
The latest commits add an example using the castle spritesheet from the displayio learn guide. The example produces this TileGrid: 
image

File: docs/api.rst [hunk 1]
```diff
@@ -12,3 +12,6 @@
 
 .. automodule:: adafruit_imageload.bmp.indexed
   :members:
+
+.. automodule:: adafruit_imageload.tilegrid_inflator
```
Comment (tekktrik) [inline docs/api.rst:16]: I always forget to do this, nice!

File: docs/examples.rst [hunk 2]
```diff
@@ -15,3 +15,12 @@ Loads image that is fetched using adafruit_request
 .. literalinclude:: ../examples/imageload_from_web.py
     :caption: examples/imageload_from_web.py
     :linenos:
+
+Inflate TileGrid test
+---------------------
+
+Load 3x3 spritesheet and inflate it to a larger sized TileGrid
+
+.. literalinclude:: ../examples/imageload_tilegrid_inflator_simpletest.py
```
Comment (tekktrik) [inline docs/examples.rst:24]: Oooo, I haven't seen this directive, I'll have to remember this one.

File: adafruit_imageload/tilegrid_inflator.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    tile_size=None,
+    transparent_index=None,
+    bmp_obj=None,
+    bmp_palette=None,
+):
+    """
+    inflate a TileGrid of ``target_size`` in tiles from a 3x3 spritesheet by duplicating
+    the center rows and columns.
+
+    :param string bmp_path: filepath to the 3x3 spritesheet bitmap file
```
[the remark this answers is not in the document]
Comment (FoamyGuy) [reply adafruit_imageload/tilegrid_inflator.py:35]: Yep, good catch. Thank you.

[the remark this answers is not in the document]
Comment (FoamyGuy) [reply adafruit_imageload/tilegrid_inflator.py]: The new commit resolves this

File: adafruit_imageload/tilegrid_inflator.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    :param tuple target_size: desired size in tiles (target_width, target_height)
+    :param Optional[tuple] tile_size: size of the tiles in the 3x3 spritesheet. If
```
[the remark this answers is not in the document]
Comment (FoamyGuy) [reply adafruit_imageload/tilegrid_inflator.py:37]: Yep, many of the others should have it as well. Lastest commit adds it to the ones that need. Thank you

**Review by tekktrik — APPROVED** [review]
Looks good to me!  Thanks @FoamyGuy!

**Pull request merged**